### PR TITLE
Multi-language CulturaX dataset configs

### PR DESCRIPTION
**Pull request by gregkocher**
## Summary:

Enable a few other datasets in other languages. This enables CulturaX dataset, specifically for Spanish (es), French (fr), German (de), and Japanese (ja).

Having other languages is helpful to bring out different biases in the model organisms. For example, this was useful for some of the Auditing Games organisms, which had certain hidden training biases specific to responses in Spanish.

Note: since this dataset is very big (~1 Pb), and for most methods we only use a very small amount of reference text data, (the first few token positions for only a few hundred or few thousand samples), I do streaming on the dataset up to the max_samples number of samples. We do NOT download any big unnecessary parquet files. After the data set is streamed over and tokenized, it is cached and can be used for model 1 and model 2 to get logits.

Datasets: [CulturaX/es](https://huggingface.co/datasets/uonlp/CulturaX/viewer/es), [CulturaX/fr](https://huggingface.co/datasets/uonlp/CulturaX/viewer/fr), [CulturaX/de](https://huggingface.co/datasets/uonlp/CulturaX/viewer/de), [CulturaX/ja](https://huggingface.co/datasets/uonlp/CulturaX/viewer/ja)

### What changed:
- `configs/config.yaml`
- `src/utils/configs.py`
- `src/diffing/methods/activation_difference_lens/act_diff_lens.py` --> ADL data loading function is edited to support "subset" and "streaming". In the Logit Diff method we import the EXACT tokenization functions from ADL to ensure parity and correctness. (Separate discussion- maybe we can move some of the loading / tokenization / apply chat template functions to a common utils.py which any method that uses teacher forcing / reference text in the methodology, could use)
- method config (I changed ADL config to use fineweb (default) and also CulturaX-es)

### Usage:
In your method config, add the following extra dataset with the subset name if needed. Example below for Spanish language (es) CulturaX, and streaming enabled:

# Example: ADL Dataset configuration:
```
datasets:
  - { id: science-of-finetuning/fineweb-1m-sample, is_chat: false, text_column: text }
  - { id: uonlp/CulturaX, is_chat: false, text_column: text, subset: es, streaming: true }
  ```

or other example, in `configs/diffing/method/logit_diff_topk_occurring.yaml` of https://github.com/science-of-finetuning/diffing-toolkit/pull/18/files#diff-85fcc2c53bd144270d9707c7f5e54ba969c32c4a8cd2f10db84579f07aeda280

### CODE REVIEW

File: src/diffing/methods/activation_difference_lens/act_diff_lens.py [hunk 1]
```diff
@@ -743,6 +755,8 @@ def compute_differences(self, dataset_entry: Dict[str, Any]) -> Dict[str, Any]:
                 text_column=dataset_entry["text_column"],
```
Comment (jkminder) [inline src/diffing/methods/activation_difference_lens/act_diff_lens.py]: I think it would also be nice to have the same for the chat datasets.
